cterasdk/core/setup.py

```diff
@@ -109,21 +109,24 @@ def init_application_server(self, ipaddr, secret):
             self._init_role(params)
         self._portal.startup.wait()
 
-    def init_replication_server(self, ipaddr, secret, replicate_from):
+    def init_replication_server(self, ipaddr, secret, replicate_from=None):
         """
         Initialize a CTERA Portal Database Replication Server.
 
         :param str ipaddr: The CTERA Portal master server IP address
         :param str secret: A password or a PEM-encoded private key
-        :param str replicate_from: The name of a CTERA Portal server replication source
+        :param str replicate_from: The name of a CTERA Portal server to replicate from
         """
         self._init_slave(ipaddr, secret)
         if self.stage == SetupWizardStage.Replication:
             logging.getLogger().info('Initializing a Replication Database Server. %s', {'host': ipaddr, 'replicate_from': replicate_from})
             logging.getLogger().debug('Retrieving database replication candidates.')
             replication_candidates = {re.search('([^/]+$)', k).group(0): k for k in self.get_replication_candidates()}
             if replication_candidates:
-                server = replication_candidates.get(replicate_from)
+                if replicate_from is None and len(replication_candidates) == 1:
```

Comment:
There are some cases that are not covered here.
1. If replicate_from is None but len is not 1 -> Shouldn't happen, but still can
2. Should we run _init_slave if we can fail later on? How can the user recover in this case? I think it's better to fail on the checks before starting anything, no?
3. Can self.stage not be SetupWizardStage.Replication? What happens if it's not?

Reply:
1. Could happen. Are you referring to the possible error scenario of where `replication_candidates > 1` and `replicate_from = None`?
2. Yes. `_init_slave` takes care of the Application Server initialization. Setting up replication is an optional step. The wizard changes its state to "Replication" after the application server is configured. To recover, the user can simply call the function again, it will just skip the app server initialization part and proceed straight to the replication.
3. If it's not "Replication" then it's finished (failed/success). That's reported by the `self._portal.startup.wait()`

Reply:
1. Yes, that's what I meant, but also len(replication_candidated) == 0 is an issue. Again users should not do this, but they might.
2. Since the API is specific for replication server, I don't see a point in running _init_slave and then failing
3. How can this happen if the user called a specific replication API?